src/shoobx/immutable/pjpersist.py

```diff
@@ -237,6 +241,11 @@ def addRevision(self, new, old=None):
         self._pj_jar.register(new)
         self._cache[new.__name__] = new
 
+    def __setitem__(self, key, value):
+        assert value.__im_state__ == interfaces.IM_STATE_LOCKED, \
+               value.__im_state__
+        super().__setitem__(key, value)
```

Comment:
There are no tests for this new code. Please add `test_setitem()` and `test_setitem_withTransientObject` to the appropriate test case.